src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine.cs

```diff
@@ -190,6 +190,8 @@ private async Task ProcessProjectAsync(Project project, ImmutableArray<ISymbol>
         using var _2 = PooledDictionary<Document, MetadataUnifyingSymbolHashSet>.GetInstance(out var documentToSymbols);
         try
         {
+            using var _3 = ArrayBuilder<Document>.GetInstance(out var documents);
```

Comment:
maybe doc semantics.  call it documentsBuffer indicating it will be reset/cleared on iterations?